packages/undo-manager/src/index.ts

```diff
@@ -0,0 +1,38 @@
+/**
+ * WordPress dependencies
+ */
+import { applyFilters } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import { createDefaultUndoManager } from './default-undo-manager';
+import { createSyncUndoManager } from './sync-undo-manager';
+
+export type { UndoManager, HistoryRecord } from './types';
+
+declare global {
+	interface Window {
+		__experimentalEnableSync?: boolean;
+	}
+}
+
+/**
+ * Creates an instance of the undo manager.
+ *
+ * Optionally, it can use a sync undo manager if the `core.useSyncUndoManager` filter is set to true.
+ *
+ * @return An instance of the undo manager.
+ */
+export function createUndoManager() {
+	let useSyncUndoManager = applyFilters( 'core.useSyncUndoManager', false );
+
+	// If collaboration hasn't been enabled, we will use the default undo manager and ignore the filter.
+	if ( ! window.__experimentalEnableSync ) {
+		useSyncUndoManager = false;
+	}
+
+	return useSyncUndoManager
+		? createSyncUndoManager()
+		: createDefaultUndoManager();
+}
```

Comment:
Instead of a boolean filter, should we use a filter that returns an UndoManager interface? Then we can put the sync undo manager in our plugin and iterate more easily.

```suggestion
export function createUndoManager() {
	let undoManager = applyFilters( 'core.undoManager', null );

	return undoManager ?? createDefaultUndoManager();
}
```

Reply:
That was the initial idea that I had. The biggest problem is that, we need to initialize the UndoManager with the doc of the post and extract the map out of that doc as well. Since the bootstrap method is present in the `provider.js` atm, that is where we would need to initialize this. We would also need to make sure we properly hook into the discard/destroy process of a document.

Considering how complicated this gets, and how tricky the undo manager already is I kept the filter as a boolean and integrated the undo manager into the existing system. 

I also figured, providing our own instance of an undo manager may not be widely accepted as we may be opening the door to some uncertainty in that area.

As a result, I think the boolean filter makes the most sense right now. We can revisit it later on.

Reply:
I am starting to think the entire sync provider should be filterable and everything housed in our plugin.

Reply:
That's the next item on my list to do, and to convert it to typescript. I agree with that

Reply:
Noting that, this will be explored later. It's complicated because of the double yjs problem, and needing to refactor core-data to work around it. That's a big 🙅🏾